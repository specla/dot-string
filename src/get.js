/**
 * Access value from object with dot-notation
 * @param  {Object|Array} item The object/Array where the value should be retreived from
 * @param  {String} path dot-notation string
 * @return {Mixed}
 */
export default function get (item, path) {
  if (typeof item !== 'object') {
    throw new TypeError('item should be an array or string')
  }

  if (typeof path !== 'string') {
    throw new TypeError('path should be a string')
  }

  return path.split('.').reduce((acc, value, index) => {
    return acc[value]
  }, item)
}
