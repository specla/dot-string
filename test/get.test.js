/* eslint-env jest */
import get from 'get'

test('should retreive value from object', () => {
  const object = { key: { item: true } }
  expect(get(object, 'key.item')).toBe(true)
})

test('should retreive value from array', () => {
  const object = [{ key: { item: true } }]
  expect(get(object, '0.key.item')).toBe(true)
})

test(`Should throw an error if the first argument isn't an array or object`, () => {
  expect(() => get('not-object', 'test')).toThrowErrorMatchingSnapshot()
})

test(`Should throw an error if the second argument isn't a string`, () => {
  expect(() => get({}, 623)).toThrowErrorMatchingSnapshot()
})
