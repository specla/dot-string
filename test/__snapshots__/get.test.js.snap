// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should throw an error if the first argument isn't an array or object 1`] = `"item should be an array or string"`;

exports[`Should throw an error if the second argument isn't a string 1`] = `"path should be a string"`;
